src/main/java/io/confluent/connect/elasticsearch/jest/JestElasticsearchClient.java

```diff
@@ -650,7 +652,26 @@ public BulkResponse executeBulk(BulkRequest bulk) throws IOException {
 
     final String errorInfo = errors.isEmpty() ? result.getErrorMessage() : errors.toString();
 
-    return BulkResponse.failure(retriable, errorInfo);
+    Map<IndexableRecord, BulkResultItem> failedResponses = new HashMap<>();
+    List<BulkResultItem> items = result.getItems();
+    for (int i = 0; i < items.size() && i < ((JestBulkRequest) bulk).records().size() ; i++) {
```

Comment:
now that `bulk` gets used in several places, maybe consider a local var at the top and casting once. 
Eg.: `JestBulkRequest jestBulk = (JestBulkRequest) bulk;` (or `jestBulkRequest` I guess)